### PR TITLE
[1LP][RFR] Fixes to ssh client connection

### PR DESCRIPTION
**Pull request by lcouzens**
Fixing tests that cause ssh connections to be left open

### CODE REVIEW

File: cfme/fixtures/cli.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def fqdn_appliance(appliance):
         raise SproutException('No provision available')
     yield apps[0]
 
+    fqdn_appliance.ssh_client.close()
```
Comment (psav) [inline cfme/fixtures/cli.py]: This won't work, as ```fqdn_appliance``` isn't defined locally here and is the fixtures name. You probably want to use ```apps[0].ssh_client.close()``` instead

File: fixtures/appliance.py [hunk 2]
```diff
@@ -32,6 +32,7 @@ def temp_appliances(count=1, preconfigured=True, lease_time=180):
             count=count, lease_time=lease_time, preconfigured=preconfigured)
         yield apps
     finally:
+        temp_appliances.ssh_client.close()
```
Comment (psav) [inline fixtures/appliance.py]: You'll need an iterator here on ```apps```

**Comment by psav** [comment]
Waiting on PRT results but looks good from here

File: cfme/tests/cli/test_appliance_cli.py [hunk 3]
```diff
@@ -4,9 +4,10 @@
 
 
 def test_set_hostname(request, appliance):
-    appliance.appliance_console_cli.set_hostname('test.example.com')
+    hostname = 'Elite-QE.redhat.com'
```
Comment (jkrocil) [inline cfme/tests/cli/test_appliance_cli.py]: Let's not.

Comment (quarckster) [inline cfme/tests/cli/test_appliance_cli.py]: 🥇

Comment (lcouzens) [inline cfme/tests/cli/test_appliance_cli.py]: @jkrocil Better?

**Pull request merged**